core/src/main/java/org/apache/accumulo/core/iterators/user/VisibilityFilter.java

```diff
@@ -46,6 +47,7 @@ public class VisibilityFilter extends Filter implements OptionDescriber {
 
   private AccessEvaluator accessEvaluator;
   protected Map<ByteSequence,Boolean> cache;
+  private ArrayByteSequence testVis = new ArrayByteSequence(new byte[0]);
```

Comment:
I think this filter was thread-safe before, but now it's not. Is that correct?

Reply:
No  this was not thread safe before and was never intended to be thread safe.  The LRUMap used for caching is not thread safe.